assets/js/data/cart/index.ts

```diff
@@ -11,19 +12,25 @@ import { STORE_KEY } from './constants';
 import * as selectors from './selectors';
 import * as actions from './actions';
 import * as resolvers from './resolvers';
-import reducer from './reducers';
+import reducer, { State } from './reducers';
 import { controls } from '../shared-controls';
 
-registerStore( STORE_KEY, {
-	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-	// @ts-ignore -- Can't figure out how to resolve this now
+registerStore< State >( STORE_KEY, {
 	reducer,
 	actions,
-	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-	// @ts-ignore -- not sure how to resolve the type issues here.
-	controls: { ...dataControls, ...controls },
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	controls: { ...dataControls, ...controls } as any,
 	selectors,
 	resolvers,
 } );
 
 export const CART_STORE_KEY = STORE_KEY;
+
+declare module '@wordpress/data' {
+	function dispatch(
+		key: typeof CART_STORE_KEY
+	): DispatchFromMap< typeof actions >;
+	function select(
+		key: typeof CART_STORE_KEY
+	): SelectFromMap< typeof selectors >;
+}
```

Comment:
Will we need to do this for collections and query-state too? Will this cause problems, or would we move this augmentation to a "central" location and just add those types to it?

Reply:
> Will we need to do this for collections and query-state too?
 Yes.

> Will this cause problems, or would we move this augmentation to a "central" location and just add those types to it?
Not sure yet. We may have to change how we export the stores and do something similar to how calypso does it for some of their data stores.